lib/puppet.rb

```diff
@@ -251,9 +251,11 @@ def self.base_context(settings)
       # doesn't exist
       default_environment = Puppet[:environment].to_sym
       if default_environment == :production
+        modulepath = settings[:modulepath]
+        modulepath = modulepath.nil? ? basemodulepath : Puppet::Node::Environment.split_path(modulepath)
         loaders << Puppet::Environments::StaticPrivate.new(
           Puppet::Node::Environment.create(default_environment,
-                                           basemodulepath,
+                                           modulepath,
```

Comment:
The documentation states:
## The base modulepath
The *base modulepath* is a list of global module directories for use with all environments. It can be configured with the `basemodulepath` setting, but its default value is probably suitable for you unless you’re doing something unusual.

The default value of the `basemodulepath` setting is `$codedir/modules:/opt/puppetlabs/puppet/modules`. (On Windows, it will just use `$codedir\modules`.)

## Using --modulepath
When running `puppet apply`, you can supply a full modulepath as a command line option. This overrides the modulepath from the current environment.

I can only interpret that as mutually exclusive.